weaver/processes/convert.py

```diff
@@ -224,29 +235,62 @@ def ows2json_io(ows_io):
-                    json_fmt_items = [j_fmt.items() for j_fmt in json_io["formats"]]
-                    if any(all(var_item in items for var_item in var_fmt.items()) for items in json_fmt_items):
+    if io_type == WPS_COMPLEX_DATA:
+        fmt_default = False
+        if "default" in json_io and isinstance(json_io["default"], dict):
+            json_io["default"]["default"] = True  # provide for workflow extension (internal), schema drops it (API)
+            fmt_default = True
+
+        # retrieve alternate format definitions
+        if "formats" not in json_io:
+            # correct complex data 'formats' from OWSLib from initial fields loop can get stored in 'supported_values'
+            fmt_val = get_field(json_io, "supported_values", pop_found=True)
+            if fmt_val:
+                json_io["formats"] = fmt_val
+            else:
+                # search for format fields directly specified in I/O body
+                for field in WPS_FIELD_FORMAT:
+                    fmt = get_field(json_io, field, search_variations=True)
+                    if not fmt:
                         continue
-                    json_io["formats"].append(var_fmt)
+                    if isinstance(fmt, dict):
+                        fmt = [fmt]
+                    fmt = filter(lambda f: isinstance(f, dict), fmt)
+                    if not isinstance(json_io.get("formats"), list):
+                        json_io["formats"] = []
+                    for var_fmt in fmt:
+                        # add it only if not exclusively provided by a previous variant
+                        json_fmt_items = [j_fmt.items() for j_fmt in json_io["formats"]]
+                        if any(all(var_item in items for var_item in var_fmt.items()) for items in json_fmt_items):
+                            continue
+                        json_io["formats"].append(var_fmt)
+                else:
+                    json_io["formats"] = []
+
+        # apply the default flag
+        for fmt in json_io["formats"]:
+            fmt["default"] = fmt_default and is_equal_formats(json_io["default"], fmt)
+            if fmt["default"]:
+                break
+
+        # fix inconsistencies of some process descriptions, both with minOccurs=1 and default format
+        if fmt_default:
+            json_io["min_occurs"] = 0
+        return json_io
+
+    # add value contrains specifications if missing
+    elif io_type in WPS_LITERAL_DATA_TYPE_NAMES:
+        domains = any2json_literal_data_domains(ows_io)
+        if domains:
+            json_io["literalDataDomains"] = domains
+            # fix inconsistencies of some process descriptions
+            # WPS are allowed to report 'minOccurs=1' although 'defaultValue' can also be provided
+            # (see https://github.com/bird-house/finch/pull/199)
+            if "defaultValue" in domains[0]:
+                json_io["min_occurs"] = 0
```

Comment:
Do this fix get shown or for internal handling only?
My interpretation is that a default value still yields a value and thus the minimum occurrence of 1 is still met.

Reply:
It is displayed in the process description to make it more obvious that this value is not mandatory and that default should be used instead. During real submit, I believe omitting it on Weaver side will omit it during the remote-WPS, which could fail. 
I have patched the relevant erroneous process though directly: https://github.com/bird-house/finch/pull/199

Reply:
Ok, given that minOccurs = 1 raises an error rather than using the default value, your fix is indeed welcome